bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxMessage.java

```diff
@@ -0,0 +1,504 @@
+                b = 0x5d;
+            }
+            frame[fi++] = b;
+        }
+
+        // 1st checksum byte
+        if (checksum1Calc == 0x7e) {
+            frame[fi++] = 0x7d;
+            frame[fi++] = 0x5e;
+        } else if (checksum1Calc == 0x7d) {
+            frame[fi++] = 0x7d;
+            frame[fi++] = 0x5d;
+        } else {
+            frame[fi++] = checksum1Calc;
+        }
+        // 2nd checksum byte
+        if (checksum2Calc == 0x7e) {
+            frame[fi++] = 0x7d;
+            frame[fi++] = 0x5e;
+        } else if (checksum2Calc == 0x7d) {
+            frame[fi++] = 0x7d;
+            frame[fi++] = 0x5d;
+        } else {
+            frame[fi++] = checksum2Calc;
+        }
+
+        return frame;
+    }
+
+    private byte[] getMessageFrameBytesInAscii() {
+        // Calculate additional bytes
+        // 1 for the start byte
+        // 2 for the length
+        // 4 for the checksum
+        // 1 for the stop byte
+        int additional = 8;
+
+        int fi = 0;
+        byte[] frame = new byte[2 * message.length + additional];
+
+        // start character
+        frame[fi++] = 0x0a;
+
+        // message length
+        String tempString = Util.byteToHex((byte) message.length);
+        frame[fi++] = (byte) tempString.charAt(0);
+        frame[fi++] = (byte) tempString.charAt(1);
+
+        // message
+        for (int i = 0; i < message.length; i++) {
+            byte b = message[i];
+            tempString = Util.byteToHex(b);
+            frame[fi++] = (byte) tempString.charAt(0);
+            frame[fi++] = (byte) tempString.charAt(1);
+        }
+
+        // Checksum 1st byte
+        tempString = Util.byteToHex(checksum1Calc);
+        frame[fi++] = (byte) tempString.charAt(0);
+        frame[fi++] = (byte) tempString.charAt(1);
+
+        // Checksum 2nd byte
+        tempString = Util.byteToHex(checksum2Calc);
+        frame[fi++] = (byte) tempString.charAt(0);
+        frame[fi++] = (byte) tempString.charAt(1);
+
+        // Stop character
+        frame[fi++] = (byte) 0x0d;
+
+        return frame;
+    }
+
+    public byte[] getMessageBytes() {
+        return message;
+    }
+
+    /**
+     * Returns a string representation of a CaddxMessage.
+     *
+     * @return CaddxMessage string
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+
+        CaddxMessageType mt = CaddxMessageType.valueOfMessageType(message[0]);
+
+        sb.append("Message: ");
+        sb.append(String.format("%2s", Integer.toHexString(message[0])));
+        sb.append(" ");
+        sb.append(mt.name);
+        sb.append("\r\n");
+
+        for (CaddxProperty p : mt.properties) {
+            sb.append("\t" + p.toString(message));
+            sb.append("\r\n");
+        }
+
+        return sb.toString();
+    }
+
+    private final Map<String, String> propertyMap = new HashMap<>();
+    private final Map<String, String> idMap = new HashMap<>();
```

Comment:
Please put all fields at the top of the file

Reply:
Done